src/BCMathExtended/BC.php

```diff
@@ -763,45 +585,39 @@ private static function recalculateNegative($number)
         return $number;
     }
 
-    /**
-     * @param string $binary
-     * @param int $base
-     * @return string
-     */
-    public static function bin2dec($binary, $base = self::MAX_BASE)
+    public static function bin2dec(string $binary, int $base = self::MAX_BASE): string
     {
         return self::decBaseHelper(
-            $base, function ($base) use ($binary) {
+            $base, static function (int $base) use ($binary) {
             $size = strlen($binary);
             $return = '0';
             for ($i = 0; $i < $size; ++$i) {
                 $element = ord($binary[$i]);
-                $power = BC::pow($base, $size - $i - 1);
-                $return = BC::add($return, BC::mul($element, $power));
+                $power = BC::pow((string)$base, (string)($size - $i - 1));
+                $return = BC::add($return, BC::mul((string)$element, $power));
             }
 
             return $return;
         }
         );
     }
 
-    /**
-     * @param string $leftOperand
-     * @param string $rightOperand
-     * @return string
-     */
-    public static function bitOr($leftOperand, $rightOperand)
+    public static function bitOr(string $leftOperand, string $rightOperand): string
     {
         return self::bitOperatorHelper($leftOperand, $rightOperand, self::BIT_OPERATOR_OR);
     }
 
-    /**
-     * @param string $leftOperand
-     * @param string $rightOperand
-     * @return string
-     */
-    public static function bitXor($leftOperand, $rightOperand)
+    public static function bitXor(string $leftOperand, string $rightOperand): string
     {
         return self::bitOperatorHelper($leftOperand, $rightOperand, self::BIT_OPERATOR_XOR);
     }
+
+    protected static function parseArgs(array $args): array
```

Comment:
Why using this `parseArgs` to check the `$args` is array type?

When using the `...$args`, this `$args` will always be the array.

And the parameters will cast type automatically if the type hint size is bigger than argument type.

Please consider following pictures to know that:

![image](https://user-images.githubusercontent.com/9021747/56636983-0b7ab300-669d-11e9-8086-5a83cc84ff3a.png)

![image](https://user-images.githubusercontent.com/9021747/56636997-17ff0b80-669d-11e9-9661-2e888a69bef2.png)

Reply:
max can handle [1,2,3] and 1,2,3 as args look \BCMathExtended\Tests\Unit\BCTest::shouldMax I didn't want to break this contract

Reply:
@krowinski. Ok. It makes sense and I see that.